packages/core/src/api/nodeUtil.test.ts

```diff
@@ -0,0 +1,361 @@
+import { describe, expect, it, vi, beforeEach, afterEach } from "vitest";
+
+import { setupTestEnv } from "./blockManipulation/setupTestEnv.js";
+import { getBlocksChangedByTransaction } from "./nodeUtil.js";
+import { Transaction } from "prosemirror-state";
+import { BlockNoteEditor } from "../editor/BlockNoteEditor.js";
+import { Step } from "prosemirror-transform";
+
+const getEditor = setupTestEnv();
+
+describe("Test getBlocksChangedByTransaction", () => {
```

Comment:
Let's add test-cases for child blocks (add / update / delete child blocks). We can debate whether this should trigger a change on the parent or not, but at least it should be tested / documented imo

Reply:
agreed should be in the tests